GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1257,15 +1319,104 @@ private string GetNewProjectedShaForPlaceholder(string path)
             return null;
         }
 
-        private void TryRemoveFolderPlaceholder(
+        private void ReExpandFolder(
+            BlobSizes.BlobSizesConnection blobSizesConnection,
+            string relativeFolderPath,
+            ConcurrentDictionary<string, PlaceholderListDatabase.PlaceholderData> updatedPlaceholderList,
+            HashSet<string> existingFolderPlaceholders)
+        {
+            FolderData folderData;
+            if (!this.TryGetOrAddFolderDataFromCache(relativeFolderPath, out folderData))
+            {
+                // Folder is no longer in the projection
+                return;
+            }
+
+            folderData.PopulateSizes(

```

Comment:
Aren't we already gathering the necessary sizes elsewhere? How do we ensure that we're not doing lots of repeated round trips to the server here?

Reply:
Filed #255 for this